dockerfiles/hpcc/templates/thor.yaml

```diff
@@ -0,0 +1,124 @@
+{{ range $thor := $.Values.thor -}}
+{{- $masterPort := $thor.masterport | default 20000 -}}
+{{- $slavePort := $thor.slaveport | default 20100 -}}
+{{- $slaveDeploymentName := printf "%s-slave" .name | quote }}
+{{- $serviceName := printf "%s-svc" .name }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .name | quote }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: {{ .name | quote }}
+  template:
+    metadata:
+      labels:
+        run: {{ .name | quote }}
+    spec:
+      containers:
+      - name: {{ .name | quote }}
+        ports:
+          - containerPort: {{ $slavePort }}
+        args: [
+                {{ include "hpcc.utils.configArg" . }},
+                {{ include "hpcc.utils.daliArg" $ }},
+                --masterport={{ $masterPort }},
+                --numSlaves={{ $thor.numSlaves }}
+              ]
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thormaster") | indent 8 }}
+{{ include "hpcc.utils.addVolumeMounts" . | indent 8 }}
+{{ include "hpcc.utils.addVolumes" . | indent 6 }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $slaveDeploymentName }}
+spec:
+  replicas: {{ $thor.numSlaves }}
+  selector:
+    matchLabels:
+      run: {{ $slaveDeploymentName }}
+  template:
+    metadata:
+      labels:
+        run: {{ $slaveDeploymentName }}
+    spec:
+      {{- if $.Values.global.singleNode | default false }}
+{{ include "hpcc.utils.changeMountPerms" (dict "volumeName" "mythorstorage" "volumePath" "/var/lib/HPCCSystems/hpcc-data") | indent 6 }}
+      {{- end }}
+      containers:
+        - name: {{ $slaveDeploymentName }}
+          args: [
+                  {{ include "hpcc.utils.configArg" . }},
+                  {{ include "hpcc.utils.daliArg" $ }},
+                  --slaveport={{ $slavePort }},
+                  --master={{ printf "%s:%v" $serviceName $masterPort }}
+                ]
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thorslave") | indent 10 }}
+          ports:
+            - containerPort: {{ $slavePort }}
+{{ include "hpcc.utils.addVolumeMounts" . | indent 10 }}
+          - mountPath: /var/lib/HPCCSystems/hpcc-data
+            name: mythorstorage
+        {{- if $thor.startDafilesrv | default false }}
+        - name: {{ printf "%s-slave-dafilesrv" .name | quote }}
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thorslave") | indent 10 }}
```

Comment:
Should there be a different image for dafilesrv?

Reply:
could be, but put here, because it is temporary, we don't want to keep dafilesrv in the loop, but for now, until the 'nas' option/functionality is available, it is necessary so that the system can 'see' the Thor file parts.